requests/src/requests/Requester.scala

```diff
@@ -230,7 +230,17 @@ case class Requester(verb: String,
         for((k, v) <- compress.headers) connection.setRequestProperty(k, v)
 
         connection.setReadTimeout(readTimeout)
-        auth.header.foreach(connection.setRequestProperty("Authorization", _))
+
+        auth match {
+          case basic: RequestAuth.Basic =>
+            connection.setRequestProperty("Authorization", basic.credentials.get)
+          case bearer: RequestAuth.Bearer =>
+            connection.setRequestProperty("Authorization", bearer.credentials.get)
```

Comment:
Line 235 and line 238 is doing the same thing, I think we can put both `Basic` and `Bearer` in the same case with `case Foo | Bar => ...`